src/pages/settings/Wallet/ReportCardLostPage.js

```diff
@@ -0,0 +1,200 @@
+import Text from '../../../components/Text';
+import MenuItemWithTopDescription from '../../../components/MenuItemWithTopDescription';
+import usePrivatePersonalDetails from '../../../hooks/usePrivatePersonalDetails';
+import assignedCardPropTypes from './assignedCardPropTypes';
+import * as CardUtils from '../../../libs/CardUtils';
+import NotFoundPage from '../../ErrorPage/NotFoundPage';
+import usePrevious from '../../../hooks/usePrevious';
+import * as FormActions from '../../../libs/actions/FormActions';
+import * as CardActions from '../../../libs/actions/Card';
+
+/** Options for reason selector */
+const OPTIONS = [
+    {
+        key: 'damaged',
+        label: 'reportCardLostOrDamaged.cardDamaged',
+    },
+    {
+        key: 'stolen',
+        label: 'reportCardLostOrDamaged.cardLostOrStolen',
+    },
+];
+
+const propTypes = {
+    /** Onyx form data */
+    formData: PropTypes.shape({
+        isLoading: PropTypes.bool,
+    }),
+    /** User's private personal details */
+    privatePersonalDetails: PropTypes.shape({
+        /** User's home address */
+        address: PropTypes.shape({
+            street: PropTypes.string,
+            city: PropTypes.string,
+            state: PropTypes.string,
+            zip: PropTypes.string,
+            country: PropTypes.string,
+        }),
+    }),
+    /** User's cards list */
+    cardList: PropTypes.objectOf(assignedCardPropTypes),
+    route: PropTypes.shape({
+        /** Each parameter passed via the URL */
+        params: PropTypes.shape({
+            /** Domain string */
+            domain: PropTypes.string,
+        }),
+    }).isRequired,
+};
+
+const defaultProps = {
+    formData: {},
+    privatePersonalDetails: {
+        address: {
+            street: '',
+            street2: '',
+            city: '',
+            state: '',
+            zip: '',
+            country: '',
+        },
+    },
+    cardList: {},
+};
+
+function ReportCardLostPage({
+    privatePersonalDetails,
+    cardList,
+    route: {
+        params: {domain},
+    },
+    formData,
+}) {
+    usePrivatePersonalDetails();
+
+    const domainCards = CardUtils.getDomainCards(cardList)[domain];
+    const physicalCard = _.find(domainCards, (card) => !card.isVirtual) || {};
+
+    const {translate} = useLocalize();
+
+    const [reason, setReason] = useState(OPTIONS[0]);
+    const [isReasonConfirmed, setIsReasonConfirmed] = useState(false);
+    const [shouldShowAddressError, setShouldShowAddressError] = useState(false);
+
+    const prevIsLoading = usePrevious(formData.isLoading);
+
+    const formattedAddress = CardUtils.getFormattedAddress(privatePersonalDetails);
+
+    useEffect(() => {
+        if (prevIsLoading && !formData.isLoading && _.isEmpty(physicalCard.errors)) {
+            Navigation.navigate(ROUTES.SETTINGS_WALLET_DOMAINCARDS.getRoute(domain));
+        }
+
+        if (formData.isLoading && _.isEmpty(physicalCard.errors)) {
```

Comment:
this should probably be the first `if` to return as early as possible

Reply:
Done!